src/reporter.js

```diff
@@ -1,60 +1,84 @@
+const utils = require("./utils.js");
+const chalk = require("chalk");
+const getPort = require("get-port");
+const WebSocket = require("ws");
+const date = require('date-and-time');
+
+let reportFilename,
+    reportDataFilename,
+    passedDataFilename,
+    smashtestSSDir,
+    initialFolder,
+    folder;
+
+const now = new Date();
+const dateFormat = date.format(now, 'YYYY-MM-DDTHH-mm-ss');
 
 /**
  * Generates a report on the status of the tree and runner
  */
 class Reporter {
-    constructor(tree, runner) {
-        this.tree = tree;               // the Tree object to report on
-        this.runner = runner;           // the Runner object to report on
+  constructor(tree, runner) {
+    this.tree = tree;               // the Tree object to report on
+    this.runner = runner;           // the Runner object to report on
 
-        this.reportTemplate = "";       // template for html reports
-        this.reportTime = null;         // Date when the report was generated
+    this.reportTemplate = "";       // template for html reports
+    this.reportTime = null;         // Date when the report was generated
 
-        this.isReportServer = true;     // whether or not to run the report server
-        this.reportDomain = null;       // domain:port where report server's api is available
-        this.wsServer = null;           // websocket server object
+    this.isReportServer = true;     // whether or not to run the report server
+    this.reportDomain = null;       // domain:port where report server's api is available
+    this.wsServer = null;           // websocket server object
 
-        this.prevSnapshot = null;       // previous snapshot sent over websockets
+    this.prevSnapshot = null;       // previous snapshot sent over websockets
 
-        this.timerFull = null;          // timer that goes off when it's time to do a full write
-        this.timerSnapshot = null;      // timer that goes off when it's time to do a snapshot write
+    this.timerFull = null;          // timer that goes off when it's time to do a full write
+    this.timerSnapshot = null;      // timer that goes off when it's time to do a snapshot write
 
-        this.stopped = false;           // true if this Reporter has been stopped
-    }
+    this.stopped = false;           // true if this Reporter has been stopped
 
-    /**
-     * @return {String} The absolute path of the report html file
-     */
+    this.reportPath = "";           // path for the report folder default is same folder smashtest file
+    this.history = false;           // activate history for reporting
+  }
+
+  /**
+   * @return {String} The absolute path of the report html file
+   */
     getFullReportPath() {
-        return path.join(process.cwd(), REPORT_FILENAME);
+        return path.join(folder, "report.html");
+    }
+
+    getPathFolder() {
+        initialFolder = (this.history === "false" ? "smashtest" : `report/smashtest-${dateFormat}`);
```

Comment:
Use `path.join()` here

Reply:
Did you mean `!this.history`?

Reply:
Can we put everything into the `smashtest` folder? So we have `smashtest/report.html`, `smashtest/report-data.js`, etc. and if the user chooses, `smashtest/reports/smashtest-[DATE]`, etc.

Reply:
Ni its this.history because is false default. but i fail false is not a string is boolean, and for your question : report.html, Report-data.js, ect. nothing has changed the architecture remains the same, just the path that is different.